addons/web_editor/static/src/js/wysiwyg/widgets/link.js

```diff
@@ -487,5 +447,51 @@ const Link = Widget.extend({
     },
 });
 
+/**
+ * Return the link element to edit. Create one from selection if none was
+ * present in selection.
+ *
+ * @param {Node} [options.containerNode]
+ * @param {Node} [options.startNode]
+ * @returns {Object}
+ */
+Link.getOrCreateLink = ({ containerNode, startNode } = {})  => {
```

Comment:
This is stable. Can't the bug be fixed without displacing the function ?

Reply:
What do you think about: 
```js
getOrCreateLink: function (editable) {
  Link.getOrCreateLink(...)
}
```

Reply:
That woud indeed solve my issue. That being said, I still don't understand the reason why you feel like you have to move the function.

Reply:
The reason is that it seems bloated to have mixed the logic that create the link inside the linktool.
I thought it to be useful to have a small refactor at the same time.

Reply:
You marked this as resolved but did not implement the method.

Reply:
Indeed, my bad.
I've added it.